sdk/auth/managed_identity_authorizer.go

```diff
@@ -169,10 +169,10 @@ func azureMetadata(ctx context.Context, url string) (body []byte, err error) {
 	req.Header = http.Header{
 		"Metadata": []string{"true"},
 	}
-	client := &http.Client{
-		Transport: http.DefaultTransport,
-		Timeout:   msiDefaultTimeout,
-	}
+
+	client := httpClient()
+	client.Timeout = msiDefaultTimeout
```

Comment:
double-checking: are there any other configuration changes we'd need to make to the retryable client given the Managed Identity API is must more limited with regards to retries than resource manager? https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#retry-guidance

Reply:
Thanks for the docs link, we should probably add a custom Backoff func for this.